game-core/src/main/java/games/strategy/triplea/attachments/TechAbilityAttachment.java

```diff
@@ -135,11 +135,13 @@ static int sumIntegerMap(final Function<TechAbilityAttachment, IntegerMap<UnitTy
   @VisibleForTesting
   static int sumNumbers(
       final ToIntFunction<TechAbilityAttachment> mapper,
+      final String attachmentType,
       final PlayerId player,
       final GameData data) {
     return TechTracker.getCurrentTechAdvances(player, data).stream()
         .map(TechAbilityAttachment::get)
         .filter(Objects::nonNull)
+        .filter(i -> i.getAttachedTo().toString().equals(attachmentType))
```

Comment:
Can you explain why this filter is needed?

Reply:
Took me a minute to think of it. The problem is that other TechAbilityAttachments for other techs get summed up which only really works if the value is zero.

I'm wondering why a nation with multiple techs doesn't have a rocket range of at least 6.